File: _CodeTopics/LeetCode/000206/000206_optm.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """

        pre = None
        curr = head

        while curr:
            next = curr.next
            curr.next = pre
            pre = curr
            curr = next 
        return pre
        
"""
https://leetcode-cn.com/submissions/detail/104875935/

27 / 27 个通过测试用例
状态：通过
执行用时: 28 ms
内存消耗: 14.9 MB
"""
"""
执行用时：28 ms, 在所有 Python 提交中击败了47.11%的用户
内存消耗：14.9 MB, 在所有 Python 提交中击败了34.45%的用户
"""

File: _CodeTopics/LeetCode/201-400/000206/tran/000206_optm.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */

struct ListNode* reverseList(struct ListNode* head){

        struct ListNode* pre = NULL;
        struct ListNode* curr = head;

        while (curr != NULL) {
            struct ListNode* next = curr->next;
            curr->next = pre;
            pre = curr;
            curr = next;
        }
        return pre;
}

"""
https://leetcode-cn.com/submissions/detail/105021785/

27 / 27 个通过测试用例
状态：通过
执行用时: 4 ms
内存消耗: 6.1 MB
"""
"""
执行用时：4 ms, 在所有 C 提交中击败了85.75%的用户
内存消耗：6.1 MB, 在所有 C 提交中击败了16.03%的用户
"""
